pkg/aws/aws.go

```diff
@@ -29,29 +30,27 @@ type awsSvc struct {
 }
 
 func New(configObject config.BaseConfig) (*awsSvc, error) {
+	var err error
 	if configObject.BucketName == "" {
 		return nil, fmt.Errorf("New: BucketName cannot be nil")
 	}
 
 	a := awsSvc{}
 
-	sess := session.Must(session.NewSessionWithOptions(session.Options{}))
-	a.awsSession = sess
-	a.ec2svc = ec2.New(sess, &aws.Config{
-		CredentialsChainVerboseErrors: aws.Bool(true),
-	})
-
+	a.awsSession = session.Must(session.NewSession())
+	a.ec2svc = a.createEC2Service(os.Getenv("AWS_REGION"))
 	a.bucketName = configObject.BucketName
-	a.s3svc = s3.New(sess)
+	a.s3svc = s3.New(a.awsSession)
 
-	err := a.getRegions()
+	err = a.getRegions()
 	if err != nil {
 		return nil, fmt.Errorf("Error Getting regions %s", err)
 	}
 
 	return &a, nil
 }
 
+//TODO: Remove a.ec2svc field and get a new service from createEC2Service
```

Comment:
The only future danger I see here is that by using `a.ec2svc` you'll get the regions of the "local account" not the "to be scanned account" and the region list may not be the same. ie If the local account disabled `us-west-1` then it wouldn't be returned, and hence instances running in `us-west-1` would never be scanned in the "to be scanned account".

Not a problem at Invoca due to us having all regions enabled, but definitely an issue someone could run into.

A workaround could be to throw down a https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#DescribeRegionsInput with `AllRegions: true` 🤷

Reply:
ditto